apps/designer/app/designer/features/style-panel/style-section.tsx

```diff
@@ -0,0 +1,777 @@
+}: // currentStyle,
+// setProperty,
+{
+  name: string;
+  currentStyle: Style;
+  setProperty: SetProperty;
+}) => {
+  return (
+    <IconButton data-property={name} css={{ width: "100%", gridArea: name }}>
+      <icons.lock.opened />
+    </IconButton>
+  );
+};
+
+const ShowMore = ({
+  styleConfigs,
+}: {
+  styleConfigs: Array<JSX.Element | null>;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  if (styleConfigs.length === 0) return null;
+  return (
+    <Collapsible.Root asChild onOpenChange={setIsOpen}>
+      <Flex direction="column" gap="3">
+        <Collapsible.Trigger asChild>
+          <Button css={{ width: "100%", gap: "$1" }}>
+            {isOpen ? <TriangleDownIcon /> : <TriangleRightIcon />}Show more
+          </Button>
+        </Collapsible.Trigger>
+        <Collapsible.Content asChild>
+          <Flex direction="column" gap="3">
+            {styleConfigs}
+          </Flex>
+        </Collapsible.Content>
+      </Flex>
+    </Collapsible.Root>
+  );
+};
+
+type RenderPropertyProps = {
+  setProperty: SetProperty;
+  currentStyle: Style;
+  inheritedStyle: InheritedStyle;
+  styleConfig: StyleConfig;
+  category: Category;
+};
+
+export const renderProperty = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+  category,
+}: RenderPropertyProps) => {
+  const Control = controls[styleConfig.control];
+  const { property } = styleConfig;
+  const key = category + "-" + property;
+  if (!Control) return null;
+  return (
+    <Box
+      key={key}
+      data-type={styleConfig.control.toLowerCase()}
+      data-property={property}
+      css={{ gridArea: property }}
+    >
+      <Control
+        currentStyle={currentStyle}
+        inheritedStyle={inheritedStyle}
+        setProperty={setProperty}
+        styleConfig={styleConfig}
+        category={category}
+      />
+    </Box>
+  );
+};
+
+type RenderCategoryProps = {
+  setProperty: SetProperty;
+  createBatchUpdate: CreateBatchUpdate;
+  currentStyle: Style;
+  category: Category;
+  styleConfigsByCategory: Array<JSX.Element | null>;
+  moreStyleConfigsByCategory: Array<JSX.Element | null>;
+};
+// Categories should render themselves because most Categories will not be flat un-ordered lists with
+// the new designs, refactor ColorControl, SpacingControl if needed.
+export const renderCategory = ({
+  setProperty,
+  createBatchUpdate,
+  currentStyle,
+  category,
+  styleConfigsByCategory,
+  moreStyleConfigsByCategory,
+}: RenderCategoryProps) => {
+  switch (category) {
+    case "layout": {
+      const css = {
+        alignItems: "center",
+        gap: "$space$styleSection",
+        "& > [data-type=iconbuttonwithmenu]": {
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          height: "100%",
+        },
+      };
+      switch (currentStyle.display?.value) {
+        case "inline-flex":
+        case "flex": {
+          return (
+            <>
+              <Grid
```

Comment:
this def. looks like it doesn't belong here

Reply:
sure, but we can move it now to make it make sense today, doesn't cost much work